basicdataset/src/main/java/ai/djl/basicdataset/ImageClassificationDataset.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.basicdataset;
+
+import ai.djl.modality.cv.Image;
+import ai.djl.modality.cv.ImageFactory;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.NDManager;
+import ai.djl.training.dataset.RandomAccessDataset;
+import ai.djl.training.dataset.Record;
+import java.io.IOException;
+
+/**
+ * A helper to create {@link ai.djl.training.dataset.Dataset}s for {@link
+ * ai.djl.Application.CV#IMAGE_CLASSIFICATION}.
+ */
+public abstract class ImageClassificationDataset extends RandomAccessDataset {
+
+    Image.Flag flag;
+
+    /**
+     * Creates a new instance of {@link RandomAccessDataset} with the given necessary
+     * configurations.
+     *
+     * @param builder a builder with the necessary configurations
+     */
+    public ImageClassificationDataset(BaseBuilder<?> builder) {
+        super(builder);
+        this.flag = builder.flag;
+    }
+
+    /**
+     * Returns the image at the given index in the dataset.
+     *
+     * @param imageFactory the imageFactory that can be used to create the {@link Image}
+     * @param index the index (if the dataset is a list of data items)
+     * @return the image
+     * @throws IOException if the image could not be loaded
+     */
+    public abstract Image getImage(ImageFactory imageFactory, long index) throws IOException;
```

Comment:
I think ImageFactory can be kept as member variable.
```suggestion
    protected abstract Image getImage(long index) throws IOException;
```